src/Altinn.Notifications/Validators/SmsNotificationOrderRequestValidator.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Text.RegularExpressions;
+
+using Altinn.Notifications.Models;
+
+using FluentValidation;
+
+namespace Altinn.Notifications.Validators;
+
+/// <summary>
+/// Class containing validation logic for the <see cref="SmsNotificationOrderRequestExt"/> model
+/// </summary>
+public class SmsNotificationOrderRequestValidator : AbstractValidator<SmsNotificationOrderRequestExt>
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SmsNotificationOrderRequestValidator"/> class.
+    /// </summary>
+    public SmsNotificationOrderRequestValidator()
+    {
+        RuleFor(order => order.Recipients)
+            .NotEmpty()
+            .WithMessage("One or more recipient is required.")
+            .Must(recipients => recipients.TrueForAll(a => IsValidMobileNumber(a.MobileNumber)))
+            .WithMessage("A valid mobile number must be provided for all recipients.");
+
+        RuleFor(order => order.RequestedSendTime)
+            .Must(sendTime => sendTime.Kind != DateTimeKind.Unspecified)
+            .WithMessage("The requested send time value must have specified a time zone.")
+            .Must(sendTime => sendTime >= DateTime.UtcNow.AddMinutes(-5))
+            .WithMessage("Send time must be in the future. Leave blank to send immediately.");
+
+        RuleFor(order => order.Body).NotEmpty();
+    }
+
+    /// <summary>
+    /// Validated as mobile number based on the Altinn 2 regex
+    /// </summary>
+    /// <param name="mobileNumber">The string to validate as an mobile number</param>
+    /// <returns>A boolean indicating that the mobile number is valid or not</returns>
+    internal static bool IsValidMobileNumber(string? mobileNumber)
+    {
```

Comment:
given a Norwegian number +47 or 0047 we want to further validate that the number starts with 9 or 4. We might also want to confirm that it is 8 digits excluding the country code, not sure if they have this last validation in Altinn 2.

Reply:
Good point! 🤔 So far I remember, Henning pointed out that the input phone number will always come validated as we are not the public facing API 🤔 Therefore we don't need that much strict validation. 🤔 So I used the Altinn 2 regex only to save time. 🤔 Please correct me if I'm wrong.

Reply:
This is a public facing API, so we should have full validation.

Reply:
This first version will take phon number as input from the service owner requesting the notification. We will in the future add the option to have Notifications do a phone number lookup. Might have been me that managed to create some confusion here, sorry.

Reply:
So what to do now? Should I spend more time on implementing those validation? Or keep it as it is? 🤔 

@SandGrainOne @acn-sbuad

Reply:
Implement validation as good as possible. Use the Altinn 2 logic as inspiration.

Reply:
This is implemented!